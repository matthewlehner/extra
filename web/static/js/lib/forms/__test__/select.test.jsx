// @flow

import React from "react";
import renderer from "react-test-renderer";

import Select from "../select";

describe("Select component", () => {
  const options = [{ value: "1", label: "Inspirational Posts" }];

  it("renders correctly", () => {
    const tree = renderer.create(
      <Select options={options} />
    );

    expect(tree).toMatchSnapshot();
  });
});
