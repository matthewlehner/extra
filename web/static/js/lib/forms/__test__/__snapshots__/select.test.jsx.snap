// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component displays placeholder 1`] = `
<div
  className="select"
>
  <select
    className="form__control"
    id="post_content[post_collection_id]"
    name="post_content[post_collection_id]"
    onChange={[Function]}
    value=""
  >
    <option>
      Select something
    </option>
    <option
      value="1"
    >
      Inspirational Posts
    </option>
  </select>
</div>
`;

exports[`Select component hides placeholder when value is present 1`] = `
<div
  className="select"
>
  <select
    className="form__control"
    id="post_content[post_collection_id]"
    name="post_content[post_collection_id]"
    onChange={[Function]}
    value="1"
  >
    <option
      value="1"
    >
      Inspirational Posts
    </option>
  </select>
</div>
`;

exports[`Select component renders correctly 1`] = `
<div
  className="select"
>
  <select
    className="form__control"
    id="post_content[post_collection_id]"
    name="post_content[post_collection_id]"
    onChange={[Function]}
    value=""
  >
    <option
      value="1"
    >
      Inspirational Posts
    </option>
  </select>
</div>
`;
